### DISCUSSION TITLE
PDF Related Record List Syntax

### DISCUSSION THREAD

**Discussion by Cattrah** [YetiForceCompany/YetiForceCRM#15942] [How to] [answered]
Could someone please clarify the "Conditions" syntax when generating a table of related records in a PDF

$(relatedRecordsList : $params)$
$params Parameter construction: RelatedModuleNameOrRelationId|Columns|**Conditions**|CustomViewIdOrName|Limit,
Example: Contacts|firstname,lastname,modifiedtime|**[[["firstname","a","Tom"]]]**||2

ie. I am creating a list of Payments In but I only want to show those in which Status="Paid" status being a picklist if that matters. I replaced "firstname" above with the fieldname in Payments in "paymentsin_status" and "Tom" with "Paid" but it doesn't return the expected results. If I wanted a list of multiple conditions, what is the syntax for that? Is it different when using other operators such as greater than/less than, is not equal, contains, etc?

Thanks

**Comment by mariuszkrzaczkowski** [comment, marked as answer]
- URL before decoding 
https://gitdeveloper.yetiforce.com/index.php?module=Contacts&view=List&viewname=7&page=1&orderby=%5B%5D&entityState=Active&mid=49&parent=47&advancedConditions=%5B%22null%22%5D&search_params=%5B%5B%5B%22firstname%22%2C%22a%22%2C%22tom%22%5D%2C%5B%22lastname%22%2C%22a%22%2C%22kow%22%5D%2C%5B%22email%22%2C%22a%22%2C%22%40yet%22%5D%5D%5D

- URL after decoding (https://www.urldecoder.org/)
https://gitdeveloper.yetiforce.com/index.php?module=Contacts&view=List&viewname=7&page=1&orderby=[]&entityState=Active&mid=49&parent=47&advancedConditions=["null"]&search_params=[[["firstname","a","tom"],["lastname","a","kow"],["email","a","@yet"]]]

what's in search_params is exactly what you're asking
[[["firstname","a","tom"],["lastname","a","kow"],["email","a","@yet"]]]

![image](https://user-images.githubusercontent.com/10198654/163527097-472d9e16-16da-4c70-80ee-38a4c407ba5f.png)

for a more technical some links
https://github.com/YetiForceCompany/YetiForceCRM/tree/developer/app/Conditions/QueryFields
https://github.com/YetiForceCompany/YetiForceCRM/blob/15657b0133a97a70177a4458aab3f16cab6abd03/app/Condition.php#L69-L102
https://github.com/YetiForceCompany/YetiForceCRM/blob/15657b0133a97a70177a4458aab3f16cab6abd03/app/Condition.php#L25-L59
https://github.com/YetiForceCompany/YetiForceCRM/blob/15657b0133a97a70177a4458aab3f16cab6abd03/app/Condition.php#L108-L140

**Reply by Cattrah** [reply]
Perfect, just what I was looking for. Thank you so much for the details!
